### PR TITLE
Migrated from gradle conventions plugin to CCUD plugin

### PR DESCRIPTION
**Pull request by ribafish**
Migrated from gradle conventions plugin to CCUD plugin and fixed and updated the Github Actions to enable CI to publish scans and write to remote build cache.

### CODE REVIEW

**Review by leonard84 — APPROVED** [review]
Same comments as in the `develocity-testing-annotations` PR

File: .github/workflows/cross-version-tests.yml [hunk 1]
```diff
@@ -5,6 +5,12 @@ on:
     branches:
     - main
   pull_request:
+  workflow_dispatch:
+
+# https://stackoverflow.com/a/72408109/16358266
```
Comment (runningcode) [inline .github/workflows/cross-version-tests.yml:10]: this seems unrelated to the goals of this PR. can you tell me more about this change?

**Review by runningcode — APPROVED** [review]

File: settings.gradle.kts [hunk 2]
```diff
@@ -10,22 +10,47 @@ pluginManagement {
...
+    }
+}
+
+buildCache {
+    local {
+        isEnabled = true
+    }
+
+    remote(develocity.buildCache) {
+        server = "https://eu-build-cache.gradle.org"
```
Comment (runningcode) [inline settings.gradle.kts:45]: why not the default?

File: .github/workflows/cross-version-tests.yml (hunk 1 above)
Comment (ribafish) [reply .github/workflows/cross-version-tests.yml:10]: Well, it allows you to run the tests manually, before opening a PR if needed.

File: settings.gradle.kts (hunk 2 above)
Comment (ribafish) [reply settings.gradle.kts:45]: See [thread](https://gradle.slack.com/archives/C032LFSKY/p1722411224756959). The default for ge.gradle.org is not used and eu node was the default till now, so that was the decision (to leave it as default).

**Review by runningcode — APPROVED** [review]

**Review by clayburn — APPROVED** [review]

**Review by leonard84 — APPROVED** [review]
